include/swift/AST/ASTContext.h

```diff
@@ -300,7 +300,9 @@ class ASTContext final {
 
     if (LangOpts.UseMalloc)
       return AlignedAlloc(bytes, alignment);
-    
+
+    if (arena == AllocationArena::Permanent && Stats)
```

Comment:
Hm, this `&& Stats` doesn't seem to be being respected. I wonder why.

Reply:
It's crashing in the ASTContext constructor, presumably before Stats is itself initialized.

Nice change otherwise!

Reply:
…I feel silly. Thanks for catching that!